packages/date-picker/src/panel/date.vue

```diff
@@ -343,9 +359,9 @@
       },
 
       confirm() {
+        this.oldDate = this.date;
```

Comment:
I thins oldDate should be set when the picker becomes visible. (See time panel's implementation). Otherwise this will happen:

![qq20180319-223006](https://user-images.githubusercontent.com/9319556/37593242-5e81a400-2bc5-11e8-91bd-34ae5010be42.gif)